testing/test_mark.py

```diff
@@ -413,6 +413,28 @@ def test_func(a, b):
     assert result.ret == 0
 
 
+def test_parametrize_iterator(testdir):
+    """parametrize should work with generators (#5354)."""
+    py_file = testdir.makepyfile(
+        """\
+        import pytest
+
+        def gen():
+            yield 1
+            yield 2
+            yield 3
+
+        @pytest.mark.parametrize('a', gen())
```

Comment:
this breaks down on marker/generator reuse

Reply:
@RonnyPfannschmidt please consider creating a new issue - or do you know what's up with the comment, @asottile ?

Reply:
I think @RonnyPfannschmidt is suggesting this won't work (not at a computer)

```python
m = pytest.mark.parametrize

@m 
def test1(): pass

@m 
def test2(): pass
```

Reply:
But if it doesn't, it never worked so :man_shrugging: probably?

Reply:
we already had a issue about reuse of generators

Reply:
yeah that example above is indeed broken, but wasn't the regression being targetted by this fix:

```console
$ pytest t.py -vv
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /tmp/venv/bin/python3
cachedir: .pytest_cache
rootdir: /tmp
plugins: celery-4.3.0
collected 4 items                                                              

t.py::test1[1] PASSED                                                    [ 25%]
t.py::test1[2] PASSED                                                    [ 50%]
t.py::test1[3] PASSED                                                    [ 75%]
t.py::test2[a0] SKIPPED                                                  [100%]

===================== 3 passed, 1 skipped in 0.01 seconds ======================
```